src/ucp/wireup/select.c

```diff
@@ -396,9 +397,20 @@ static UCS_F_NOINLINE ucs_status_t ucp_wireup_select_transport(
             local_iface_flags |= UCT_IFACE_FLAG_CONNECT_TO_IFACE;
         }
 
+        local_md_flags = criteria->local_md_flags;
+        if (!(iface_attr->cap.flags & UCT_IFACE_FLAG_CONNECT_TO_IFACE) ||
+            /* If the transport has both CONNECT_TO_IFACE and CONNECT_TO_EP, and
+             * connecting via CM, UCP prefers CONNECT_TO_EP */
+            ((iface_attr->cap.flags & UCT_IFACE_FLAG_CONNECT_TO_EP) &&
+             (ucp_ep_init_flags_has_cm(select_params->ep_init_flags)))) {
+            /* Remove INVALIDATE md flag in case of transport uses CONNECT_TO_EP
+             * connection mode */
+            local_md_flags &= ~UCT_MD_FLAG_INVALIDATE;
+        }
```

Comment:
why need to remove flag? better not to add in first place if not needed..

Reply:
will be done in the separate PR